From a8427ce51f8b4a31fa122844db8ceec9c728a1ac Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Fri, 26 Sep 2014 09:16:39 +0200
Subject: [PATCH] dialog: fixed typo/copy&paste oversight to dlg pointer name
 in clean routine

- credits to Alex Balshov for report, troubleshooting and spotting the
  fix

(cherry picked from commit 8e6c83182677ec234c5f44c5dfcb40e421a21bcb)
---
 modules/dialog/dlg_hash.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules/dialog/dlg_hash.c b/modules/dialog/dlg_hash.c
index 8b94b01..4284801 100644
--- a/modules/dialog/dlg_hash.c
+++ b/modules/dialog/dlg_hash.c
@@ -250,11 +250,11 @@ int dlg_clean_run(ticks_t ti)
 				destroy_dlg(tdlg);
 			}
 			if(tdlg->state==DLG_STATE_CONFIRMED_NA && tdlg->start_ts<tm-60) {
-				if(update_dlg_timer(&dlg->tl, 10)<0) {
+				if(update_dlg_timer(&tdlg->tl, 10)<0) {
 					LM_ERR("failed to update dialog lifetime in long non-ack state\n");
 				}
-				dlg->lifetime = 10;
-				dlg->dflags |= DLG_FLAG_CHANGED;
+				tdlg->lifetime = 10;
+				tdlg->dflags |= DLG_FLAG_CHANGED;
 			}
 		}
 		lock_set_release(d_table->locks, d_table->entries[i].lock_idx);
-- 
2.1.1

